### PR TITLE
[DOCS] Removes Data Connector references

### PR DESCRIPTION
**Pull request by Rachel-Reverie**
## Details
- Removes Data Connector from the glossary
- Removes usages of Data Connector as a technical term
- Removes references to Data Connectors in the docs and updates them according to the Fluent Datasource workflow.

## Definition of done
- [X] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Appropriate tests and docs have been updated

### CODE REVIEW

File: docs/docusaurus/docs/glossary.md [hunk 1]
```diff
@@ -25,8 +25,6 @@ import CLIRemoval from '/docs/components/warnings/_cli_removal.md'
 
 [**Data Assistant:**](./terms/data_assistant.md) A utility that asks questions about your data, gathering information to describe what is observed, and then presents Metrics and proposes Expectations based on the answers.
 
-[**Data Connector:**](./terms/data_connector.md) Provides the configuration details based on the source data system which are needed by a Datasource to define Data Assets.
```
Comment (Kilo59) [inline docs/docusaurus/docs/glossary.md:28]: Is it worth keeping the entry in the glossary but saying something like "DataConnectors are now an implementation detail that users don't need to care about"?

File: docs/docusaurus/docs/integrations/integration_zenml.md [hunk 2]
```diff
@@ -36,8 +36,7 @@ tools and services that spans everything else MLOps.
 
 ZenML ships with a couple of builtin pipeline steps that take care of everything
 from configuring temporary <TechnicalTag tag="datasource" text="Datasources" />,
-<TechnicalTag tag="data_connector" text="Data Connectors" />, 
-and <TechnicalTag tag="batch_request" text="Runtime Batch Requests" /> 
+Data Connectors, and <TechnicalTag tag="batch_request" text="Runtime Batch Requests" />
```
Comment (Kilo59) [inline docs/docusaurus/docs/integrations/integration_zenml.md:39]: Should this reference be removed too?

**Review by Kilo59 — APPROVED** [review]

**Comment by tjholsman** [comment]
@Rachel-Reverie should we add this doc to the sidebar? Or is this an older one?

https://deploy-preview-7930--niobium-lead-7998.netlify.app/docs/guides/connecting_to_your_data/how_to_choose_between_working_with_a_single_or_multiple_batches_of_data/

**Comment by Rachel-Reverie** [comment]
> @Rachel-Reverie should we add this doc to the sidebar? Or is this an older one?

This is an older document.  I think it still has some value, but we should review and update it so that it fits in better in the context of FDS workflows.  It may even be best to convert it to a conceptual guide, rather than leaving it in the form of a how-to.

**Review by tjholsman — APPROVED** [review]

**Pull request merged**